<launch>
  <arg name="nodelet_manager_name" value="nodelet_manager_leftcam_top" />
  <arg name="camera_name" value="leftcam_top" />

  <node pkg="nodelet" type="nodelet" name="$(arg nodelet_manager_name)" args="manager" output="screen" />
  
  <node pkg="nodelet" type="nodelet" name="ueye_cam_nodelet_leftcam_top"
        args="load ueye_cam/ueye_cam_nodelet $(arg nodelet_manager_name)">


    <param name="camera_name" type="str" value="$(arg camera_name)" /> 
    <param name="camera_topic" type="str" value="image_raw" />
    <param name="camera_id" type="int" value="1" /> 
 

    <param name="ext_trigger_mode" type="bool" value="True" /> 
    <param name="color_mode" type="str" value="bgr8" />
   
    <param name="image_width" type="int" value="1936" />
    <param name="image_height" type="int" value="1216" />
    <param name="image_top" type="int" value="-1" /> <!-- -1: center -->
    <param name="image_left" type="int" value="-1" /> <!-- -1: center -->

    <param name="subsampling" type="int" value="1" /> <!-- supported by only some UEye cameras -->
    <param name="binning" type="int" value="1" /> <!-- supported by only some UEye cameras -->
    <param name="sensor_scaling" type="double" value="1.0" /> <!-- supported by only some UEye cameras -->

    <param name="auto_gain" type="bool" value="False" />
    <param name="master_gain" type="int" value="0" />
    <param name="red_gain" type="int" value="30" />
    <param name="green_gain" type="int" value="0" />
    <param name="blue_gain" type="int" value="50" />
    <param name="gain_boost" type="bool" value="False" />

    <param name="auto_exposure" type="bool" value="False" />
    <param name="exposure" type="int" value="33" /> <!-- in ms -->

    <param name="auto_white_balance" type="bool" value="False" />
    <param name="white_balance_red_offset" type="int" value="0" />
    <param name="white_balance_blue_offset" type="int" value="0" />
    
    
    <param name="auto_frame_rate" type="bool" value="False" />
    <param name="frame_rate" type="double" value="12.0" />
    <param name="pixel_clock" type="int" value="40" />

    <param name="flip_lr"  type="bool" value="True" />
  </node>
</launch>
